### PR TITLE
test_stepstoreproduce.py: Add basic tests

### PR DESCRIPTION
**Pull request by sladyn98**
Added basic tests with respec to the
stepstoreproduce model.

Reference: https://github.com/mozilla/bugbug/issues/428

### CODE REVIEW

**Comment by sladyn98** [comment]
@marco-c Im sorry this is the first time reading the log of TaskCluster. I am not able to read where exaclty is the log. Could you help me a bit

**Comment by marco-c** [comment]
On this page, click on Details. You will see a list of tasks, the failing ones are in red and say "failed". Click on the name of one of them. Click on "Run Logs", select one of the two options.

Note: you should be able to reproduce the errors locally too.

**Comment by sladyn98** [comment]
Got it :+1: And to run this repo I just need to execute `run.py` as mentioned in the readme.Is it necessary to run the mining scripts?

**Comment by marco-c** [comment]
@sladyn98 so you didn't try to run the test that you wrote?

You don't need to run the training nor the mining scripts, you just need to run the test that you added.

**Comment by sladyn98** [comment]
@marco-c I did run the test and I got a key error. I suppose thats not what the classes dictionary expects.Thats why the tests are failing

File: tests/test_stepstoreproduce.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+from bugbug.models.stepstoreproduce import StepsToReproduceModel
+
+
+def test_get_labels():
+    model = StepsToReproduceModel()
+    classes, _ = model.get_labels()
+    # assert not classes[1101825]
+    assert classes[1042096]
```
Comment (ayush-1506) [inline tests/test_stepstoreproduce.py]: @sladyn98 The stepstoreproduce model doesn't have this key in it's labels.

**Comment by ayush-1506** [comment]
@sladyn98 Both these models are independent of each other. You should look at a bug that satisfies the `stepstoreproduce` criteria (found in the get_labels method) and add a check to see if the result is consistent.

**Comment by sladyn98** [comment]
Is there a way to find a key ,such that the key exists?I mean would I have to run the script...
cc @ayush-1506 , @marco-c

**Comment by ayush-1506** [comment]
No. All you need is the bugs.json file. You don't need to run the repository mining script.

**Comment by sladyn98** [comment]
@marco-c @ayush-1506 Done Thanks and kindly review.

**Comment by sladyn98** [comment]
I did try adding a negative test but there were no bugs for which the label was 0. This was the only bug.

**Comment by ayush-1506** [comment]
That shouldn't happen. Can you tell me what calling the `get_labels` method for this classifier returns for you? How large is your bugs DB file?

**Comment by sladyn98** [comment]
I just ran pytest and printed the classes.
```
0 bugs have no steps to reproduce
1 bugs have steps to reproduce
{1488310: 1}
```
Doesnt the bugs db for tests get populated from the json?

**Comment by ayush-1506** [comment]
Yes, I'm guessing you don't have the complete bugs.json file for some reason. What is the size of data/bugs.json ?

**Comment by sladyn98** [comment]
Its showing me `385.1 kB (3,85,144 bytes) ` on my linux system

**Comment by ayush-1506** [comment]
Yes, it should be around 1.4 GBs. Delete the data folder and run `python run.py --goal=component --train` (You can use and possible value for goal, this will just download the bugs and commits DB).

**Comment by ayush-1506** [comment]
Or you can directly download it from https://github.com/mozilla/bugbug/blob/master/bugbug/bugzilla.py#L21 and paste it in the data folder.

**Comment by sladyn98** [comment]
Yeah I did keep it for download :)

**Comment by sladyn98** [comment]
@ayush-1506 The tests pass on my local machine.But they are failing on task cluster

**Comment by sladyn98** [comment]
I'm guessing it's because the tests are being picked up from the mock dB and the key is not present there.

**Comment by marco-c** [comment]
> I'm guessing it's because the tests are being picked up from the mock dB and the key is not present there.

Yes, you need to add it to the mock DB.

**Comment by sladyn98** [comment]
And the bugs are not properly formatted in the json file which is making it difficult to read. Could you tell me the fields in a bug along with the start and end field. Thanks

**Comment by marco-c** [comment]
> And the bugs are not properly formatted in the json file which is making it difficult to read. Could you tell me the fields in a bug along with the start and end field. Thanks

You'll need to make a bit of effort yourself to solve the issue, otherwise it's useless for you (you don't learn much) and it is useless for me (as I spend more time explaining every small detail to you than to just fix the issue myself).

**Comment by sladyn98** [comment]
Yeah agreed. Will come up with a solution soon. Thanks

**Comment by sladyn98** [comment]
@marco-c Kindly have a look I added some mock data.The tests are passing too.

File: tests/test_stepstoreproduce.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+
+from bugbug.models.stepstoreproduce import StepsToReproduceModel
+
+
+def test_get_labels():
+    model = StepsToReproduceModel()
+    classes, _ = model.get_labels()
+    assert classes[1488310]
+    assert not classes[1372243]
```
Comment (marco-c) [inline tests/test_stepstoreproduce.py]: Could you also add a bug which is not in classes (you can use one of the bugs which is already in the mock DB and has `cf_has_str` not set to `yes` and not set to `no`).

Comment (marco-c) [reply tests/test_stepstoreproduce.py]: assert XXX not in classes

Comment (sladyn98) [reply tests/test_stepstoreproduce.py]: You mean choose a bug that has `cf_has_str : ---- ` and assert that it is not in classes?

Comment (marco-c) [reply tests/test_stepstoreproduce.py]: Yes!

Comment (sladyn98) [reply tests/test_stepstoreproduce.py]: Done

Comment (sladyn98) [reply tests/test_stepstoreproduce.py]: Done kindly review.This should suffice IMO :)

**Pull request merged**